checks/branch_protected.go

```diff
@@ -54,55 +55,87 @@ func IsBranchProtected(protection *github.Protection, c *checker.CheckRequest) c
+	} else {
+		totalSuccess++
 	}
 
-	if protection.GetEnforceAdmins() != nil {
-		if !protection.EnforceAdmins.Enabled {
-			c.Logf("!! branch protection EnforceAdmins not enabled")
-		} else {
-			totalSuccess++
-		}
+	// This is disabled by default (bad).
+	if protection.GetEnforceAdmins() != nil &&
+		protection.EnforceAdmins.Enabled {
+		totalSuccess++
+	} else {
+		c.Logf("!! branch protection EnforceAdmins not enabled")
 	}
 
-	if protection.GetRequireLinearHistory() != nil {
-		if !protection.RequireLinearHistory.Enabled {
-			c.Logf("!! branch protection require linear history not enabled")
-		} else {
-			totalSuccess++
-		}
+	// This is disabled by default (bad).
+	if protection.GetRequireLinearHistory() != nil &&
+		protection.RequireLinearHistory.Enabled {
+		totalSuccess++
+	} else {
+		c.Logf("!! branch protection require linear history not enabled")
 	}
 
-	if protection.GetRequiredStatusChecks() != nil {
+	// This is disabled by default (bad).
+	if protection.GetRequiredStatusChecks() != nil &&
+		protection.RequiredStatusChecks.Strict &&
+		len(protection.RequiredStatusChecks.Contexts) > 0 {
+		totalSuccess++
+	} else {
 		switch {
-		case !protection.RequiredStatusChecks.Strict:
+		case protection.RequiredStatusChecks == nil ||
+			!protection.RequiredStatusChecks.Strict:
 			c.Logf("!! branch protection require status checks to pass before merging not enabled")
 		case len(protection.RequiredStatusChecks.Contexts) == 0:
 			c.Logf("!! branch protection require status checks to pass before merging has no specific status to check for")
 		default:
-			totalSuccess++
+			panic("unhandled status check error")
 		}
 	}
 
-	if protection.GetRequiredPullRequestReviews() != nil {
-		if protection.RequiredPullRequestReviews.RequiredApprovingReviewCount < 1 {
+	// This is disabled by default (bad).
+	if protection.GetRequiredPullRequestReviews() != nil &&
+		protection.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews &&
+		protection.RequiredPullRequestReviews.DismissStaleReviews &&
+		(protection.RequiredPullRequestReviews.DismissalRestrictions == nil ||
+			(protection.RequiredPullRequestReviews.DismissalRestrictions.Users == nil &&
+				protection.RequiredPullRequestReviews.DismissalRestrictions.Teams == nil)) &&
+		protection.RequiredPullRequestReviews.RequireCodeOwnerReviews {
+		totalSuccess++
+
+	} else {
+		switch {
+		case protection.RequiredPullRequestReviews == nil:
 			c.Logf("!! branch protection require pullrequest before merging not enabled")
-		} else {
-			totalSuccess++
+			fallthrough
+		case protection.RequiredPullRequestReviews.RequiredApprovingReviewCount < minReviews:
+			c.Logf("!! branch protection require %v pullrequest reviews before merging not enabled", minReviews)
+			fallthrough
+		case !protection.RequiredPullRequestReviews.DismissStaleReviews:
+			c.Logf("!! branch protection require dismissal stale reviews before merging not enabled")
```

Comment:
such an unreadable error message, since you are here, lets improve them all

!! branch protection rule - dismiss stale reviews on new commits should be enabled
or something like that ?